sdk/servicebus/service-bus/src/receivers/shared.ts

```diff
@@ -43,3 +44,20 @@ export async function* getMessageIterator<ReceivedMessageT>(
     yield messages[0];
   }
 }
+
+/**
+ * @internal
+ * @ignore
+ */
+export function wrapProcessErrorHandler(
+  handlers: Pick<MessageHandlers<unknown>, "processError">,
+  logError: (formatter: any, ...args: any[]) => void = log.error
+): MessageHandlers<unknown>["processError"] {
+  return async (err: Error) => {
+    try {
+      await handlers.processError(err);
+    } catch (err) {
+      logError(`An error was thrown from the user's processError handler: ${err}`);
+    }
+  };
```

Comment:
Minor, you can take it or leave it, but this doesn't have to be an async function:
```suggestion
  return ((err: Error) => handlers.processError(err).catch((err) => {
      logError(`An error was thrown from the user's processError handler: ${err}`);
    })
  )};
```

Reply:
I'm inclined to leave this as-is mostly because I think the style is a bit easier to read (totally 100% subjective on my part).